File: src/main/java/top/zbeboy/isy/elastic/pojo/StaffElastic.java
package top.zbeboy.isy.elastic.pojo;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import org.apache.commons.lang3.math.NumberUtils;
import org.springframework.data.annotation.Id;
import org.springframework.data.elasticsearch.annotations.Document;

import java.sql.Date;

/**
 * Created by lenovo on 2017-04-12.
 */
@Document(indexName = "staff", type = "staff", shards = 1, replicas = 0, refreshInterval = "-1")
@NoArgsConstructor
@ToString
public class StaffElastic {
    @Id
    private String staffId;
    @Getter
    @Setter
    private String staffNumber;
    @Getter
    @Setter
    private Date birthday;
    @Getter
    @Setter
    private String sex;
    @Getter
    @Setter
    private String idCard;
    @Getter
    @Setter
    private String familyResidence;
    @Getter
    @Setter
    private Integer politicalLandscapeId;
    @Getter
    @Setter
    private String politicalLandscapeName;
    @Getter
    @Setter
    private Integer nationId;
    @Getter
    @Setter
    private String nationName;
    @Getter
    @Setter
    private Integer academicTitleId;
    @Getter
    @Setter
    private String academicTitleName;
    @Getter
    @Setter
    private String post;
    @Getter
    @Setter
    private Integer schoolId;
    @Getter
    @Setter
    private String schoolName;
    @Getter
    @Setter
    private Integer collegeId;
    @Getter
    @Setter
    private String collegeName;
    @Getter
    @Setter
    private Integer departmentId;
    @Getter
    @Setter
    private String departmentName;
    @Getter
    @Setter
    private String username;
    @Getter
    @Setter
    private Byte enabled;
    @Getter
    @Setter
    private String realName;
    @Getter
    @Setter
    private String mobile;
    @Getter
    @Setter
    private String avatar;
    @Getter
    @Setter
    private String langKey;
    @Getter
    @Setter
    private Date joinDate;
    /**
     * -1 : 无权限
     * 0 :  有权限
     * 1 : 系统
     * 2 : 管理员
     */
    @Getter
    @Setter
    private Integer authorities;
    /**
     * 以空格分隔的角色名
     */
    @Getter
    @Setter
    private String roleName;

    public Integer getStaffId() {
        return NumberUtils.toInt(staffId);
    }

    public void setStaffId(Integer staffId) {
        this.staffId = staffId + "";
    }
}

File: src/main/java/top/zbeboy/isy/elastic/pojo/StaffElastic.kt
package top.zbeboy.isy.elastic.pojo

import org.apache.commons.lang.math.NumberUtils
import org.springframework.data.annotation.Id
import org.springframework.data.elasticsearch.annotations.Document
import java.sql.Date

/**
 * Created by zbeboy 2017-11-19 .
 **/
@Document(indexName = "staff", type = "staff", shards = 1, replicas = 0, refreshInterval = "-1")
open class StaffElastic {
    @Id
    private var staffId: String? = null
    var staffNumber: String? = null
    var birthday: String? = null
    var sex: String? = null
    var idCard: String? = null
    var familyResidence: String? = null
    var politicalLandscapeId: Int? = null
    var politicalLandscapeName: String? = null
    var nationId: Int? = null
    var nationName: String? = null
    var academicTitleId: Int? = null
    var academicTitleName: String? = null
    var post: String? = null
    var schoolId: Int? = null
    var schoolName: String? = null
    var collegeId: Int? = null
    var collegeName: String? = null
    var departmentId: Int? = null
    var departmentName: String? = null
    var username: String? = null
    var enabled: Byte? = null
    var realName: String? = null
    var mobile: String? = null
    var avatar: String? = null
    var verifyMailbox: Byte? = null
    var langKey: String? = null
    var joinDate: Date? = null
    /**
     * 详见：ElasticBook
     */
    var authorities: Int? = null
    /**
     * 以空格分隔的角色名
     */
    var roleName: String? = null

    constructor()

    constructor(staffId: String?, staffNumber: String?, birthday: String?, sex: String?, idCard: String?, familyResidence: String?, politicalLandscapeId: Int?, politicalLandscapeName: String?, nationId: Int?, nationName: String?, academicTitleId: Int?, academicTitleName: String?, post: String?, schoolId: Int?, schoolName: String?, collegeId: Int?, collegeName: String?, departmentId: Int?, departmentName: String?, username: String?, enabled: Byte?, realName: String?, mobile: String?, avatar: String?, verifyMailbox: Byte?, langKey: String?, joinDate: Date?, authorities: Int?, roleName: String?) {
        this.staffId = staffId
        this.staffNumber = staffNumber
        this.birthday = birthday
        this.sex = sex
        this.idCard = idCard
        this.familyResidence = familyResidence
        this.politicalLandscapeId = politicalLandscapeId
        this.politicalLandscapeName = politicalLandscapeName
        this.nationId = nationId
        this.nationName = nationName
        this.academicTitleId = academicTitleId
        this.academicTitleName = academicTitleName
        this.post = post
        this.schoolId = schoolId
        this.schoolName = schoolName
        this.collegeId = collegeId
        this.collegeName = collegeName
        this.departmentId = departmentId
        this.departmentName = departmentName
        this.username = username
        this.enabled = enabled
        this.realName = realName
        this.mobile = mobile
        this.avatar = avatar
        this.verifyMailbox = verifyMailbox
        this.langKey = langKey
        this.joinDate = joinDate
        this.authorities = authorities
        this.roleName = roleName
    }

    fun getStaffId(): Int? {
        return NumberUtils.toInt(staffId)
    }

    fun setStaffId(staffId: Int?) {
        this.staffId = staffId!!.toString() + ""
    }
}